role_test.go

```diff
@@ -57,6 +58,37 @@ const (
 	deleteRoleResponse  = `{"message":"Role deleted"}`
 )
 
+func TestRoles(t *testing.T) {
+	mockData := strings.Repeat(getRoleResponse+",", 1000) // make 1000 roles.
+	mockData = "[" + mockData[:len(mockData)-1] + "]"     // remove trailing comma; make a json list.
+
+	// This creates 1000 + 1000 + 1 (2001, 3 calls) worth of roles.
```

Comment:
Can we also test what happens if 0 or any number below 1,000 is returned?

Reply:
Both of them are tested with separate test cases with my new commit. Just a comment though. The same test is made  for folders and dashboards and these cases are not covered. More or less I copied and modified the logic of those tests.

Hopefully we are good to go now 😄

Reply:
Yup, that's better! I'm aware some other tests do not cover all cases (yaaay) but given that we are adding new tests I think it's better if we test them thoroughly.